src/main/java/org/broadinstitute/hellbender/engine/spark/SAMRecordToGATKReadAdapterSerializer.java

```diff
@@ -86,12 +82,12 @@ public LazyBAMRecord(
 
         @Override
         public void setReferenceIndex(final int referenceIndex) {
```

Comment:
Can we delete this `LazyBAMRecord` class (and corresponding factory)? I don't think it's needed any longer, as it's fine to call `setReferenceIndex(SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX)` on a headerless read. 

Keeping this around will cause problems, I think, since when we write out our reads, the writer will now auto-restore the header, and it will expect get/set reference index methods to function properly.